main.py

```diff
@@ -1,31 +1,73 @@
+import asyncio
 import argparse
 from vid2persona import init
 from vid2persona.pipeline import vlm
+from vid2persona.pipeline import llm
 
 def validate_args(func):
     def inner_function(args):
+        gcp_project_id, gcp_project_location = init.get_env_vars()
+
+        if args.gcp_project_id is None: args.gcp_project_id = gcp_project_id
+        if args.gcp_project_location is None: args.gcp_project_location = gcp_project_location
+
         if args.gcp_project_id is None or args.gcp_project_location is None:
             raise ValueError("gcp-project-id or gcp-project-location is missing")
+        
+        if args.hf_access_token is not None:
+            # https://huggingface.co/blog/inference-pro
+            allowed_models_hf_pro = [
+                "mistralai/Mixtral-8x7B-Instruct-v0.1",
+                "NousResearch/Nous-Hermes-2-Mixtral-8x7B-DPO",
+                "mistralai/Mistral-7B-Instruct-v0.2",
+                "mistralai/Mistral-7B-Instruct-v0.1",
+                "HuggingFaceH4/zephyr-7b-beta",
+                "meta-llama/Llama-2-7b-chat-hf",
+                "meta-llama/Llama-2-13b-chat-hf",
+                "meta-llama/Llama-2-70b-chat-hf",
+                "openchat/openchat-3.5-0106"
+            ]
+            if args.model_id not in allowed_models_hf_pro:
+                raise ValueError("not supported model for Hugging Face PRO account")
         return func(args)
     return inner_function
 
 @validate_args
-def workflow(args):
```

Comment:
Seems like this function is for convenience as it doesn't return anything.

Reply:
yes. the `main.py` CLI is to verify if other parts are working fine as a library.